File: MusicPlayer.v2/app/src/main/java/com/example/musicplayerv2/MainActivity.java
package com.example.musicplayerv2;

import android.content.Intent;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity implements Callback, SharedPreferences.OnSharedPreferenceChangeListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        setTheme(ThemePicker.getTheme(getApplicationContext()));
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        SongAdapter songAdapter = new SongAdapter(new AppListDiffCallback(), this);
        RecyclerView recyclerView = findViewById(R.id.rv_songs);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(linearLayoutManager);
        recyclerView.setAdapter(songAdapter);
        songAdapter.submitList(getSongs());
        SharedPreferences preferences =
                PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
        preferences.registerOnSharedPreferenceChangeListener(this);
    }

    @Override
    public void exactSong(int position) {
        Intent intent = new Intent(this, PlayActivity.class);
        intent.putExtra("position", position);
        startActivity(intent);
    }

    @Override
    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
        recreate();
    }

    public static ArrayList<Song> getSongs() {
        ArrayList<Song> songs = new ArrayList<>();
        songs.add(new Song(R.raw.song_1, "First Song", "First Artist"));
        songs.add(new Song(R.raw.song_2, "Second Song", "Second Artist"));
        songs.add(new Song(R.raw.song_3, "Third Song", "Third Artist"));
        songs.add(new Song(R.raw.song_4, "Fourth Song", "Fourth Artist"));
        songs.add(new Song(R.raw.song_1, "Fifth Song", "Fifth Artist"));
        return songs;
    }
}

File: MusicPlayer.v2Kotlin/app/src/main/java/com/example/musicplayerv2/MainActivity.kt
package com.example.musicplayerv2

import android.content.Intent
import android.content.SharedPreferences
import android.preference.PreferenceManager
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView

import java.util.ArrayList

class MainActivity : AppCompatActivity(), Callback, SharedPreferences.OnSharedPreferenceChangeListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        setTheme(ThemePicker.getTheme(applicationContext))
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val songAdapter = SongAdapter(AppListDiffCallback(), this)
        val recyclerView = findViewById<RecyclerView>(R.id.rv_songs)
        val linearLayoutManager = LinearLayoutManager(this)
        recyclerView.layoutManager = linearLayoutManager
        recyclerView.adapter = songAdapter
        songAdapter.submitList(songs)
        val preferences = PreferenceManager.getDefaultSharedPreferences(applicationContext)
        preferences.registerOnSharedPreferenceChangeListener(this)
    }

    override fun exactSong(position: Int) {
        val intent = Intent(this, PlayActivity::class.java)
        intent.putExtra("position", position)
        startActivity(intent)
    }

    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences, key: String) {
        recreate()
    }

    companion object {

        val songs: ArrayList<Song>
            get() {
                val songs = ArrayList<Song>()
                songs.add(Song(R.raw.song_1, "First Song", "First Artist"))
                songs.add(Song(R.raw.song_2, "Second Song", "Second Artist"))
                songs.add(Song(R.raw.song_3, "Third Song", "Third Artist"))
                songs.add(Song(R.raw.song_4, "Fourth Song", "Fourth Artist"))
                songs.add(Song(R.raw.song_1, "Fifth Song", "Fifth Artist"))
                return songs
            }
    }
}
